# Generated by Django 4.1b1 on 2022-07-08 11:07

from django.db import migrations, models


class Migration(migrations.Migration):

    dependencies = [
        ("core", "0011_rename_category_order_to_sort_order_step1"),
    ]

    operations = [
        migrations.AddConstraint(
            model_name="category",
            constraint=models.UniqueConstraint(
                condition=models.Q(("parent__isnull", True)),
                fields=("project", "name"),
                name="core_category_name_no_parent_uniq",
                violation_error_message="A category with this name already exists in this project.",
            ),
        ),
        migrations.AddConstraint(
            model_name="category",
            constraint=models.UniqueConstraint(
                condition=models.Q(("parent__isnull", False)),
                fields=("parent", "name"),
                name="core_category_name_parent_uniq",
                violation_error_message="A sub-category with this name already exists in this category.",
            ),
        ),
        migrations.AddConstraint(
            model_name="category",
            constraint=models.UniqueConstraint(
                condition=models.Q(("parent__isnull", True)),
                fields=("project", "sort_order"),
                name="core_category_order_no_parent_uniq",
                violation_error_message="A category with this order already exists in this project.",
            ),
        ),
        migrations.AddConstraint(
            model_name="category",
            constraint=models.UniqueConstraint(
                condition=models.Q(("parent__isnull", False)),
                fields=("parent", "sort_order"),
                name="core_category_order_parent_uniq",
                violation_error_message="A sub-category with this order already exists in this category.",
            ),
        ),
    ]
